hyperlink/_url.py

```diff
@@ -970,7 +950,9 @@ def from_text(cls, text):
                             raise URLParseError('port must not be empty')
                         raise URLParseError('expected integer for port, not %r'
                                             % port_str)
-        family, host = parse_host(host)
+        # _, host = parse_host(host)
+        if host:
+            host = host.lstrip('[').rstrip(']')
```

Comment:
Could this just be represented as a modification to the RE?  I would prefer to see it there than here, because here it seems like potentially invalid uses of `[` might sneak in.  (For example, a host that begins `[[[[[`).  If the `host` group only contained the appropriate text it would be cleaner to read as well.